src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Class responsible for displaying the user interface of Duke.
+ */
+public class Ui {
+
+    private Scanner scanner;
+
+    /**
+     * Constructor for Ui.
+     */
+    public Ui() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Displays the current taskList to the user.
+     *
+     * @param tl the taskList to be shown to the user.
+     * @return String containing display message to user.
+     */
+    public String showList(TaskList tl) {
+        StringBuilder sb = new StringBuilder();
+        List<Task> taskList = tl.getTasks();
+        int index = 1;
+        sb.append("Here are the tasks in your list nya:\n");
+        for (Task t : taskList) {
+            sb.append(index + "." + t + "\n");
+            index++;
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Displays the current taskList to the user.
+     *
+     * @param tl the taskList to be shown to the user.
+     * @param filter the filter to find tasks by.
+     * @return String containing display message to user.
+     */
+    public String showList(TaskList tl, String filter) {
+        StringBuilder sb = new StringBuilder();
+        List<Task> taskList = tl.getTasks();
+        int index = 1;
+        sb.append("Here are the matching tasks in your list nya:\n");
+        for (Task t : taskList) {
+            if (t.description.contains(filter)) {
+                sb.append(index + "." + t + "\n");
+                index++;
+            }
+        }
+        return sb.toString();
```

Comment:
Is it better to move the logic of getting the relevant tasks to TaskList class instead? So as to follow the "tell, don't ask" principle.